src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -267,10 +267,10 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* c
   // has updated oops.
   StackWatermarkSet::after_unwind(current);
 
-  cm = CodeCache::find_compiled(pc);
-  assert(cm != nullptr, "this is not a compiled method");
+  nm = CodeCache::find_nmethod(pc);
+  assert(nm != nullptr, "this is not a compiled method");
```

Comment:
Unrelated to this patch, but might be worth mentioning because I think it would be good to think about this in a follow-up RFE. `CodeCache::find_blob` returns null if it can't find a matching blob, but `CodeCache::find_nmethod` asserts that it did find one. The latter makes the assert redundant, but it also begs to question if `find_blob` and `find_nmethod` realy should be different in this regard? Should we have `find_blob_or_null` and `find_nmethod_or_null`? Alt. `find_blob_not_null` and `find_nmethod_not_null`.